src/org/traccar/model/Position.java

```diff
@@ -65,15 +66,45 @@
     public static final String PREFIX_ADC = "adc";
     public static final String PREFIX_IO = "io";
     public static final String PREFIX_COUNT = "count";
+    public static final String PREFIX_BUTTON = "button";
 
+    public static final String ENDING_PRESSED = "Pressed";
+    public static final String ENDING_RELEASED = "Released";
+    public static final String ENDING_OPEN = "Open";
+    public static final String ENDING_CLOSE = "Close";
+    public static final String ENDING_ACTIVE = "Active";
+    public static final String ENDING_INACTIVE = "Inactive";
+
+
+    public static final String ALARM_GENERAL = "general";
     public static final String ALARM_SOS = "sos";
     public static final String ALARM_VIBRATION = "vibration";
     public static final String ALARM_MOVEMENT = "movement";
     public static final String ALARM_OVERSPEED = "overspeed";
+    public static final String ALARM_OVERSPEED_END = "overspeedEnd";
     public static final String ALARM_FALL_DOWN = "fallDown";
     public static final String ALARM_LOW_BATTERY = "lowBattery";
     public static final String ALARM_MOTION = "motion";
     public static final String ALARM_FAULT = "fault";
+    public static final String ALARM_POWER_OFF = "powerOff";
+    public static final String ALARM_JAMMING = "jamming";
+    public static final String ALARM_GEOFENCE = "geofence";
+    public static final String ALARM_GEOFENCE_ENTER = "geofenceEnter";
+    public static final String ALARM_GEOFENCE_EXIT = "geofenceExit";
+    public static final String ALARM_ODB = "odb";
```

Comment:
I think it shouldn't be alarm. We already handle OBD DTC codes for some other protocols.